libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/HistoryRecordingStateChangeObserver.kt

```diff
@@ -0,0 +1,56 @@
+package com.mapbox.navigation.core.internal
+
+import androidx.annotation.UiThread
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+
+/**
+ * Interface definition for an observer that gets notified whenever
+ * some actions are required for history recording.
```

Comment:
```suggestion
 * Interface definition for an observer that gets notified whenever
 * some actions are required to record each trip session (Free Drive and Active Guidance) independently.
```
We can follow up separately to avoid CI cycles blocking the release.